DSMAJCEexamples/src/main/java/DSMAJCE.java

```diff
@@ -0,0 +1,119 @@
+    private static String FORTANIX_API_ENDPOINT = "<Endpoint URL>";
+
+    // replace with your api-key
+    private static String FORTANIX_API_KEY = "<API Key>";
+    // replace with your app uuid. Only required for cert-based auth
+    private static String FORTANIX_APP_UUID = "<APP UUID>";
+
+    // for config-file or env variable usage
+    private static SdkmsJCE provider;
+    //api-key auth
+    private static SdkmsJCE provider1;
+    //cert-based auth
+    private static SdkmsJCE provider2;
+
+    private static boolean encrypt_decrypt() throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidAlgorithmParameterException, InvalidKeyException, IllegalBlockSizeException, BadPaddingException {
+        String algorithm = String.format("AES_256/CBC/NOPADDING", 256, CBC, NOPADDING);
+        Cipher cipher = Cipher.getInstance(algorithm, provider);
+        SobjectDescriptor sobjDesc = new SobjectDescriptor().kid("<Key UUID>");
+        KeyObject keyObj = SdkmsKeyService.getKeyObject(sobjDesc);
+        SdkmsAESKey key = new SdkmsAESKey(keyObj);
+        AlgorithmParameters params = cipher.getParameters();
+        cipher.init(Cipher.ENCRYPT_MODE, key, params);
+        String PLAIN = "the small brown fox jumps over the lazy dog";;
+        System.out.println("Plain text: " + PLAIN);
+        // Encrypted content
+        byte[] cipherBytes = cipher.doFinal(PLAIN.getBytes());
+        System.out.println("CBC mode | encrypted to " + new String(cipherBytes));
+        // decrypt the same content
+        cipher.init(Cipher.DECRYPT_MODE, key, params);
+        byte[] plainBytes = cipher.doFinal(cipherBytes);
+        System.out.println("CBC mode | decrypted to " + new String(plainBytes));
+        return Arrays.equals(PLAIN.getBytes(), plainBytes);
+    }
+
+    private static boolean tokenization() throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidAlgorithmParameterException, InvalidKeyException, IllegalBlockSizeException, BadPaddingException {
+        String algorithm = String.format("AES_256/FPE/NOPADDING", 256, FPE, NOPADDING);
+        Cipher cipher = Cipher.getInstance(algorithm, provider1);
+        SobjectDescriptor sobjDesc = new SobjectDescriptor().name("<Key Name>");
+        KeyObject keyObj = SdkmsKeyService.getKeyObject(sobjDesc);
+        SdkmsAESKey key = new SdkmsAESKey(keyObj);
+        AlgorithmParameters params = cipher.getParameters();
+        cipher.init(Cipher.ENCRYPT_MODE, key, params);
+        String PLAIN = "5200828282828210";
+        System.out.println("Plain text: " + PLAIN);
+        // Encrypted content
+        byte[] cipherBytes = cipher.doFinal(PLAIN.getBytes());
+        System.out.println("FPE mode | tokenized to " + new String(cipherBytes));
+        // decrypt the same content
+        cipher.init(Cipher.DECRYPT_MODE, key, params);
+        byte[] plainBytes = cipher.doFinal(cipherBytes);
+        System.out.println("FPE mode | detokenized to " + new String(plainBytes));
```

Comment:
We could / should show using an App with masking enabled versus an App without masking enabled.  In the comments, we should give direction to create the tokenization sobj, a then a masked / non masked App.  Then, in this code to detokenize, we should use both the masked and unmasked App to demonstrate masking.